com.unity.render-pipelines.universal/Runtime/UniversalRenderPipelineGlobalSettings.cs

```diff
@@ -0,0 +1,240 @@
+using UnityEditor;
+
+namespace UnityEngine.Rendering.Universal
+{
+    /// <summary>
+    /// Universal Render Pipeline's Global Settings.
+    /// Global settings are unique per Render Pipeline type. In URP, Global Settings contain:
+    /// - light layer names
+    /// </summary>
+    partial class UniversalRenderPipelineGlobalSettings : RenderPipelineGlobalSettings
+    {
+        #region Version system
+        [SerializeField] int k_AssetVersion = 1;
+        [SerializeField] int k_AssetPreviousVersion = 1;
+
+        public void OnAfterDeserialize()
+        {
+#if UNITY_EDITOR
+            if (k_AssetPreviousVersion != k_AssetVersion)
+            {
+                EditorApplication.delayCall += () => UpgradeAsset(this);
+            }
+#endif
+        }
+
+#if UNITY_EDITOR
+        static void UpgradeAsset(UniversalRenderPipelineGlobalSettings asset)
+        {
+            EditorUtility.SetDirty(asset);
+        }
+
+#endif
+        #endregion
+
+        private static UniversalRenderPipelineGlobalSettings cachedInstance = null;
+        public static UniversalRenderPipelineGlobalSettings instance
+        {
+            get
+            {
+                if (cachedInstance == null)
+                    cachedInstance = GraphicsSettings.GetSettingsForRenderPipeline<UniversalRenderPipeline>() as UniversalRenderPipelineGlobalSettings;
+                return cachedInstance;
+            }
+        }
+
+        static internal void UpdateGraphicsSettings(UniversalRenderPipelineGlobalSettings newSettings)
+        {
+            if (newSettings == null || newSettings == cachedInstance)
+                return;
+            GraphicsSettings.RegisterRenderPipelineSettings<UniversalRenderPipeline>(newSettings as RenderPipelineGlobalSettings);
+            cachedInstance = newSettings;
+        }
+
+#if UNITY_EDITOR
+        //Making sure there is at least one UniversalGlobalSettings instance in the project
+        internal static UniversalRenderPipelineGlobalSettings Ensure(string folderPath = "", bool canCreateNewAsset = true)
+        {
+            if (UniversalRenderPipelineGlobalSettings.instance)
+                return UniversalRenderPipelineGlobalSettings.instance;
+
+            UniversalRenderPipelineGlobalSettings assetCreated = null;
+            string path = "Assets/" + folderPath + "/UniversalRenderPipelineGlobalSettings.asset";
```

Comment:
This is a hard coded string that is sued twice. Maybe save it somewhere?

Reply:
i am not sure to understand your comment. what string is used twice exactly?

Reply:
In UniversalGlobalSettingsCreator file, there is also this
/UniversalRenderPipelineGlobalSettings.asset
Maybe that name could come from a file instead? So only from one source. I don't know if we have a utils file for these kind of things. (I checked and I couldnt find) But maybe we should? :) 
Maybe in UniversalRenderPipelineAsset?

Reply:
A pattern I see in SRP is to load resource asset by GUId instead of path string. This is safer and don't break on renaming.